functions/helpFunctions.tsx

```diff
@@ -99,3 +99,52 @@ export function clearLocalStorage(): void {
     localStorage.clear();
   }
 }
+
+// b = true will return outliers among those score with a high freq
+export function chooseFrequentAssessmentBasedOnScore(assessments : AssessmentType[], b : boolean) {
+  const numberOfAGivenScore = Array.from({length: assessments[0].maxPoints+1}, () => 0); // number of times x points are given, points = index
+  var hasNullScore = false;                      
+  assessments.map((assessment) => {
+    assessment.score == null ? hasNullScore = true :
+    numberOfAGivenScore[assessment.score] += 1
+  });
+
+  var n : number;
+  if (b) {
+    n = Math.max(...numberOfAGivenScore); 
+  } else {
+    const reducedNumberOfAGivenScore = numberOfAGivenScore.filter(n => n <= 0);
+    n = Math.min(...reducedNumberOfAGivenScore);
+  }
+    const res : number[] = [];
+    numberOfAGivenScore.forEach((item, index) => item === n ? res.push(index): null);
+
+    const score = res[Math.floor(Math.random() * res.length)];
+    const assessmentWithMostFrequentScore =  assessments.filter(assessment => assessment.score == score);
```

Comment:
This might need a name as sometimes the outlier is the one with least frequencies?